### PR TITLE
[tchserver-gauntlet PART1] - Implement gauntlet pure-tch server

### PR DESCRIPTION
**Pull request by breerly**
@yarpc/golang

### CODE REVIEW

File: crossdock/server/tch/gauntlet.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+func (thriftTestHandler) TestException(ctx thrift.Context, arg string) error {
+	ctx.SetResponseHeaders(ctx.Headers())
+	switch arg {
+	case "Xception":
+		code := int32(1001)
+		return &gauntlet_apache.Xception{ErrorCode: &code, Message: &arg}
+	case "TException":
+		// TODO is there something better I can raise here?
+		// unexpected exception.
+		return errors.New("great sadness")
```
Comment (breerly) [inline crossdock/server/tch/gauntlet.go]: @prashantv does tchannel-go have a `TException`-like thing?

Comment (prashantv) [inline crossdock/server/tch/gauntlet.go]: don't need to do anything special, any unknown error (e.g. not a thrift exception declared on this endpoint) will be handled as an unknown error in TChannel

Comment (breerly) [inline crossdock/server/tch/gauntlet.go]: Roger - I'll remove the TODO then

File: crossdock/server/tch/server.go [hunk 2]
```diff
@@ -65,6 +66,8 @@ func register(ch *tchannel.Channel) {
 	ch.Register(raw.Wrap(echoRawHandler{}), "echo/raw")
 	json.Register(ch, json.Handlers{"echo": echoJSONHandler}, onError)
 	thrift.NewServer(ch).Register(echo.NewTChanEchoServer(&echoThriftHandler{}))
+	thrift.NewServer(ch).Register(gauntlet_apache.NewTChanThriftTestServer(&thriftTestHandler{}))
+	thrift.NewServer(ch).Register(gauntlet_apache.NewTChanSecondServiceServer(&secondServiceHandler{}))
```
Comment (abhinav) [inline crossdock/server/tch/server.go]: Discussed offline but: Re-use `thrift.NewServer(ch)` for all registers.

**Pull request merged**